automation/taskcluster/lib/tasks.py

```diff
@@ -170,27 +311,248 @@ def craft_default_task_definition(
+                'groupSymbol': _craft_treeherder_group_symbol_from_variant(variant) if variant else None,
+                'jobKind': 'other',
+                'machine': {
+                  'platform': 'android-all' if is_nightly else _craft_treeherder_platform_from_variant(variant),
+                },
+                'symbol': 'Ns' if is_nightly else 'As',
+                'tier': 1,
+            },
+        )
+
+    def _craft_nightly_routes(self, is_nightly):
+        routes = []
+
+        if is_nightly:
+            index_release = 'staging-signed-nightly' if is_staging else 'signed-nightly'
+            routes = [
+                "index.project.mobile.reference-browser.{}.nightly.{}.{}.{}.latest".format(
+                    index_release, self.date.year, self.date.month, self.date.day
+                ),
+                "index.project.mobile.reference-browser.{}.nightly.{}.{}.{}.revision.{}".format(
+                    index_release, self.date.year, self.date.month, self.date.day, self.commit
+                ),
+                "index.project.mobile.reference-browser.{}.nightly.latest".format(index_release),
+            ]
+
+        return routes
+
+    def craft_push_task(self, signing_task_id, is_staging=True):
+        payload = {
+            "commit": True,
+            "google_play_track": 'nightly',
+            "upstreamArtifacts": [
+                {
+                    "paths": _APKS_PATHS,
+                    "taskId": signing_task_id,
+                    "taskType": "signing"
+                }
+            ]
+        }
+
+        return self._craft_default_task_definition(
+            worker_type='mobile-pushapk-dep-v1' if is_staging else 'mobile-pushapk-v1',
+            provisioner_id='scriptworker-prov-v1',
+            dependencies=[signing_task_id],
+            routes=[],
+            scopes=[
+                "project:mobile:reference-browser:releng:googleplay:product:reference-browser{}".format(
+                    ':dep' if is_staging else ''
+                )
+            ],
+            name="Push task",
+            description="Upload signed release builds of Reference-Browser to Google Play",
+            payload=payload,
+            treeherder={
+                'jobKind': 'other',
+                'machine': {
+                  'platform': 'android-all',
+                },
+                'symbol': 'gp',
+                'tier': 1,
+            },
+        )
+
+    def craft_upload_apk_nimbledroid_task(self, assemble_task_id):
+        # For GeckoView, upload nightly (it has release config) by default, all Release builds have WV
+        return self._craft_build_ish_task(
+            name="Upload Release APK to Nimbledroid",
+            description='Upload APKs to Nimbledroid for performance measurement and tracking.',
+            command=' && '.format([
+                'wget https://queue.taskcluster.net/v1/task/{}/artifacts/public/target.arm.apk'.format(assemble_task_id),
```

Comment:
Hey @kglazko ! @mitchhentges and I are refactoring decision-nightly.py. I seized this opportunity to stop the nimbledroid task from rebuilding a release APK. Instead, it fetches it from the upstream task. Please let me know if this is okay to you.

Reply:
FYI, this was tested in staging at https://tools.taskcluster.net/groups/N3cuLfTxQxWZZk7diQYG1g/tasks/YuLCobrNQcKWzRjpWiktfw/runs/0/logs/public%2Flogs%2Flive_backing.log#L397

Reply:
Sounds great to me- that should cut down the task time :)

Reply:
Awesome ! 😃